environment_secrets.go

```diff
@@ -0,0 +1,98 @@
+/*
```

Comment:
Would it be clearer when the file was named something like `application_marshaling.go`? In the end it defines the marshalling / unmarshalling of a application. WDYT?